### ISSUE TITLE
Load More button on profile posts view unavailable for certain users

### ISSUE THREAD

**Issue by jordanjay29** [flarum/core#1681]
## Bug Report

**Current Behavior**
On [luceos' profile at discuss](https://discuss.flarum.org/u/luceos), the posts view no longer has a Load More button at the bottom. Other user profile pages do not appear to be affected. Perhaps his number of posts (almost 6000) is too much for Flarum's components? 

**Expected Behavior**
Should have a Load More button as is customary. 

**Environment**
discuss.flarum.org

**Comment by luceos** [comment]
Went over this with @jordanjay29 just now. Yesterday I created a draft (made private with byobu) on discuss about releasing premium extensions to the public. After some feedback I felt it was better to have the discussion removed and hard deleted it.

It seems the post in the database is still intact; for people with db access: postId: 88569, discussionId: 17869. It seems posts aren't deleted.

I'm not quite sure why this affects the posts lists, but that's the only thing that came to mind.

Now what I also found in the UserPostsPage component is this:

```js
    if (this.loading) {
      footer = LoadingIndicator.component();
    } else if (this.moreResults) {
      footer = (
        <div className="PostsUserPage-loadMore">
          {Button.component({
            children: app.translator.trans('core.forum.user.posts_load_more_button'),
            className: 'Button',
            onclick: this.loadMore.bind(this)
          })}
        </div>
      );
    }
```

and 

```js
this.moreResults = results.length >= this.loadLimit;
```

So whenever the results is larger than the load limit there are more results, but shouldn't something like that rely on a pagination links attribute next?

https://jsonapi.org/format/#fetching-pagination
